datadog/resource_datadog_user.go

```diff
@@ -196,8 +199,28 @@ func resourceDatadogUserCreate(ctx context.Context, d *schema.ResourceData, meta
 			if existingUser == nil {
 				return diag.Errorf("could not find single user with email %s", email)
 			}
-		} else {
-			existingUser = &responseData[0]
+		}
+		if existingUser == nil {
+			// We already raised an exception if multiple users were found but no exact email match.
+			// If user is nil at this stage, we can assume a user with the same handle already exists.
+			// Find the user and raise a helpful error message.
```

Comment:
i see there is a check for statuscode 409 at the top, why not return the error message for user already exists with handle there?